modules/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureBlobStore.java

```diff
@@ -231,95 +237,100 @@ public boolean blobExists(OperationPurpose purpose, String blob) throws IOExcept
-     * @see <a href="https://github.com/elastic/elasticsearch/pull/65140#discussion_r528752070">Discussion on implementing PR</a>
-     */
-    @Override
-    public void deleteBlobsIgnoringIfNotExists(OperationPurpose purpose, Iterator<String> blobs) {
-        if (blobs.hasNext() == false) {
-            return;
+    private void deleteListOfBlobs(AzureBlobServiceClient azureBlobServiceClient, Iterator<String> blobNames) throws IOException {
+        // We need to use a container-scoped BlobBatchClient, so the restype=container parameter
+        // is sent, and we can support all SAS token types
+        // See https://learn.microsoft.com/en-us/rest/api/storageservices/blob-batch?tabs=shared-access-signatures#authorization
+        final BlobBatchAsyncClient batchAsyncClient = new BlobBatchClientBuilder(
+            azureBlobServiceClient.getAsyncClient().getBlobContainerAsyncClient(container)
+        ).buildAsyncClient();
+        final List<Mono<Void>> batchResponses = new ArrayList<>();
+        while (blobNames.hasNext()) {
+            final BlobBatch currentBatch = batchAsyncClient.getBlobBatch();
+            int counter = 0;
+            while (counter < maxDeletesPerBatch && blobNames.hasNext()) {
+                currentBatch.deleteBlob(container, blobNames.next());
+                counter++;
+            }
+            batchResponses.add(batchAsyncClient.submitBatch(currentBatch));
         }
 
-        BlobServiceAsyncClient asyncClient = asyncClient(purpose);
-        SocketAccess.doPrivilegedVoidException(() -> {
-            final BlobContainerAsyncClient blobContainerClient = asyncClient.getBlobContainerAsyncClient(container);
-            try {
-                Flux.fromStream(StreamSupport.stream(Spliterators.spliteratorUnknownSize(blobs, Spliterator.ORDERED), false))
-                    .flatMap(blob -> getDeleteTask(blob, blobContainerClient.getBlobAsyncClient(blob)), CONCURRENT_DELETES)
-                    .then()
-                    .block();
-            } catch (Exception e) {
-                filterDeleteExceptionsAndRethrow(e, new IOException("Unable to delete blobs"));
+        try {
+            Mono.whenDelayError(
+                batchResponses.stream()
+                    .map(m -> m.onErrorResume(AzureBlobStore::isIgnorableBatchDeleteException, t -> Mono.empty()))
+                    .toList()
+            ).block();
+        } catch (Exception e) {
+            final IOException exception = new IOException("Error deleting batch", e);
+            if (Exceptions.isMultiple(e)) {
+                Exceptions.unwrapMultipleExcludingTracebacks(e)
+                    .stream()
+                    // This is horrific, but there's no way to filter this noise reliably, see BlockingSingleSubscriber#blockingGet
+                    .filter(throwable -> throwable.getMessage().contains("#block terminated with an error") == false)
```

Comment:
I really don't like this, but for some reason if we use `#block()` to wait for a Mono to complete, we end up with a suppressed `Exception` that has the message `#block terminated with an error` in the list of suppressed exceptions. It's added [here](https://github.com/reactor/reactor-core/blob/dd0fbdcf87915ba4529918962e38b1981facef53/reactor-core/src/main/java/reactor/core/publisher/BlockingSingleSubscriber.java#L146) for some reason.

Some other options I considered are
- Include the exception we caught here as a cause and don't manipulate it at all (leaving the noisy exception in)
- Copy over the suppressed exceptions unfiltered (leaving the noisy exception in)
- Write our own implementation of `#block()`/`BlockingMonoSubscriber` that doesn't add the noise in - this gets pretty boiler-plate-y

I'm open to suggestions about what to do here.
-

Reply:
I ended up using the subscribe approach